### PR TITLE
[Server] Use filename as design name while importing design

### LINKED ISSUE #11026: [Server] Design Import: Use filename as design name
### Current Behavior
During design import (of any supported file type), when no name for the design is provided, "Autogenerated" is used as the design name.

### Expected Behavior
Using the filename of as the Design name is more descriptive and helpful.

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by dusdjhyeon**
**Notes for Reviewers**

This PR fixes #11026 

- [x] Add NewPatternFileFromK8sManifest parameter to get file name (Without that parameter, it will still be set to “Autogenerated”. )
- [x] Get the file name with 'getElementById' on the front end (since the [schema](https://github.com/meshery/meshkit) comes from meshkit, it was hard to modify it directly).

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Yay, @dusdjhyeon! 🎈

**Review by shubham251972 — COMMENTED** [review]
@dusdjhyeon thanks for raising this PR, you are making sure that if file name is "dapr.yaml" we are using "dapr" as name of design and removing file extension from name right?

**Comment by dusdjhyeon** [comment]
> @dusdjhyeon thanks for raising this PR, you are making sure that if file name is "dapr.yaml" we are using "dapr" as name of design and removing file extension from name right?

Right! Remove the extension with split before sending it to the backend.

**Review by MUzairS15 — COMMENTED** [review]
https://github.com/meshery/meshery/pull/11097/files#diff-e1d76eb522ad8ca1078d26e59d9c2635f97eeeaefdd95f7c693683fe3e151bd5R1532
@dusdjhyeon Let's use the `name` (if provided)during URL imports.

**Comment by dusdjhyeon** [comment]
@MUzairS15 
Thank you for your review! I updated.
If you imported "https://helm.traefik.io/mesh/traefik-mesh-3.0.6.tgz", You should get "traefik-mesh-3.0.6". 
But the problem arises when it's a .tar.gz file. Please tell me if I should consider this case as well, because filepath.Ext uses LastIndexOf('.").

Also, I have a question. 
Why does githubRepoDesignScan only pass over filenames with the [extension removed](https://github.com/meshery/meshery/blob/5cc6b44032879918dcb112e1bb778596fa401921/server/handlers/meshery_pattern_handler.go#L846), while,
genericHTTPDesignFile is passed in with the extension as [url[len(url)-1]](https://github.com/meshery/meshery/blob/5cc6b44032879918dcb112e1bb778596fa401921/server/handlers/meshery_pattern_handler.go#L910)?

**Comment by leecalcote** [comment]
@Yashsharma1911, you might have some insight to offer.

**Comment by vishalvivekm** [comment]
Thank you @dusdjhyeon 
are you available to discuss this item in Meshery Dev meeting starting in 20 minutes ?
Meeting minutes: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#
Meeting link: https://meet.layer5.io/meshery

**Comment by leecalcote** [comment]
@dusdjhyeon did we get this PR finalized?

**Comment by dusdjhyeon** [comment]
> @dusdjhyeon did we get this PR finalized?
Yes!

**Review by leecalcote — DISMISSED** [review]

File: ui/components/MesheryPatterns.js [hunk 1]
```diff
@@ -1513,15 +1513,19 @@ function MesheryPatterns({
     const { uploadType, name, url, file, designType } = data;
     let requestBody = null;
     switch (uploadType) {
-      case 'File Upload':
+      case 'File Upload': {
```
Comment (leecalcote) [inline ui/components/MesheryPatterns.js:1516]: Ideally, this is case insensitive.

Comment (dusdjhyeon) [reply ui/components/MesheryPatterns.js:1516]: ok, I understand. I could use a 'toLowerCase()' integration, but the uploadType is already listed [here](https://github.com/meshery/meshery/blob/7b3def9f13ec53f46b67fe0f116f4dfafda45cef/ui/components/MesheryPatterns.js#L1519) as one of two. Do I need to modify that?

**Review by MUzairS15 — APPROVED** [review]

**Pull request merged**